saskatoon/sitebase/templates/app/detail_views/harvest/about.html

```diff
@@ -8,7 +9,7 @@
                     <div class="curved-ctn">
                         <h2>{% trans "Public announcement" %}</h2>
                         <hr><br>
-                        <h5>{% for t in harvest.trees %} {{ t }},{% endfor %}
+                        <h5>{% for tree in trees %} {{ tree.name }} {{ tree.fruit_name|get_fruit_name_icon }},{% endfor %}
```

Comment:
same here, I don't think you make use of your serializer but of the `harvest.trees.all()` objects. If your custom template tag works for `property.trees` it should now also work with looping over `harvest.trees`

Reply:
> 



> Thanks for the great work @mina-andajos! Yes I agree with @tristanlatr we could remove the colored squares.
> 
> @mina-andajos if you're interested I would like to suggest an improvement based on the modification you made.In `HarvestViewset` I am still passing `trees': harvest.trees.all()` to the Response, mainly because I needed the `tree.id` for the fruit distribution form: https://github.com/LesFruitsDefendus/saskatoon-ng/blob/master/saskatoon/sitebase/templates/app/detail_views/harvest/distribution/tree_select.html Ideally I think we should use the new serializer you just created and get the trees with `{% for t in harvest.trees %}`. There might be another adjustment needed. If it sounds clear to you maybe let's open another Issue/PR. Thanks!

If you recall the discussion on https://github.com/LesFruitsDefendus/saskatoon-ng/discussions/144, the `HarvestSerializer` isn't used in `HarvestViewset.retrieve()` which is used for the harvest detail views. So I believe we will serialize all the data there to apply what you're suggesting? Or can I just use the `TreeTypeSerializer` just for the trees?

Reply:
@mina-andajos  Actually I believe `HarvestSerializer` is used in `'harvest': response.data` when calling `response = super(HarvestViewset, self).retrieve(request, pk=pk)` (through the definition of `serializer_class`) so you should now be able to access your nested TreeTypeSerializer in `harvest.trees`. Do you mind check that?

Reply:
Thanks for pointing this out. I'm working on it right now